### ISSUE TITLE
Issues with PIC code on latest release

### ISSUE THREAD

**Issue by NewbiZ** [seladb/PcapPlusPlus#1216]
Hello,

I have issues when linking the latest release static libraries (23.09) to a shared library. 
```
/usr/bin/ld: /usr/local/lib/libCommon++.a(SystemUtils.cpp.o): warning: relocation against `_ZN4pcpp23ApplicationEventHandler14handlerRoutineEi' in read-only section `.text'
/usr/bin/ld: /usr/local/lib/libPcap++.a(PcapFileDevice.cpp.o): relocation R_X86_64_PC32 against symbol `_ZN4pcpp22PcapNgFileWriterDevice11writePacketERKNS_9RawPacketE' can not be used when making a shared object; recompile with -fPIC
```

I did not spend too much time investigating yet, but it looks to me the new CMake build system does not generate PIC unless when building shared libraries.
In my case, I would like the pre-built static libraries of PcapPlusPlus to be PIC, as they used to be in 22.11.

I don't think there is any downside to _always_ generate PIC, and it makes life easier for users creating shared libraries.

**Comment by seladb** [comment]
In general, PcapPlusPlus doesn't support shared libraries because some of its APIs don't work on some platforms.

Adding @clementperon who worked on the new CMake build system

**Comment by clementperon** [comment]
@NewbiZ thanks for your report.

You could set this easily with Cmake just add
set(POSITION_INDEPENDENT_CODE TRUE)

And rebuild it, I will double check what are the impacts but I think you re right we should build static librairies as PIC.

**Comment by clementperon** [comment]
@NewbiZ Thanks for the report I have opened a MR

**Comment by NewbiZ** [comment]
@clementperon that looks great, thanks for making the change!

Do you want me to check the build before release?

**Comment by clementperon** [comment]
Fixed with #1219

**Comment by avallee-eisler** [comment]
Hello @clementperon,
Would you be able to create a release with this change included?
